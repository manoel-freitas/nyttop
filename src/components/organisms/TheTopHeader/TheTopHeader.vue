<template>
  <ny-level :extra-classes="extraClasses">
    <ny-level-item v-if="$mq === 'laptop' || $mq === 'desktop'" side="left">
      <ny-level-item>
        <ny-icon icon="bars" @click="$emit('onOpenSideMenu')" />
      </ny-level-item>
    </ny-level-item>
    <ny-level-item v-if="$mq === 'mobile' || $mq === 'tablet'" side="left">
      <ny-level-item>
        <ny-icon icon="bars" @click="$emit('onOpenSideMenu')" />
      </ny-level-item>
    </ny-level-item>

    <ny-level-item v-if="$mq === 'laptop' || $mq === 'desktop'">
      <h1 class="is-size-3">The New York Top</h1>
    </ny-level-item>
    <ny-level-item v-if="$mq === 'mobile' || $mq === 'tablet'" side="right">
      <h1 class="is-size-3">NYTTOP</h1>
    </ny-level-item>
  </ny-level>
</template>

<script>
export default {
  computed: {
    extraClasses() {
      return (this.$mq === 'laptop') | (this.$mq === 'desktop')
        ? ''
        : 'is-mobile'
    }
  }
}
</script>

<style></style>
