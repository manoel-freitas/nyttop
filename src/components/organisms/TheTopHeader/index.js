import TheTopHeader from './TheTopHeader'

export default TheTopHeader
