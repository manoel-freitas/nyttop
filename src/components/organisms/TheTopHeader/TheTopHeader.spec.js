/// <reference types="jest" />

import { shallowMount } from '@vue/test-utils'
import TheTopHeader from './TheTopHeader'

let theTopHeader = shallowMount(TheTopHeader, {
  stubs: ['ny-level', 'ny-level-item', 'ny-icon']
})
describe('TheTopHeader Component', () => {
  test('TheTopHeader has to be mounted', () => {
    expect(theTopHeader.isVueInstance()).toBeTruthy()
  })

  test('snapshot test', () => {
    expect(theTopHeader.element).toMatchSnapshot()
  })

  test('should extraClasses to be `is-mobile`', () => {
    const $mq = 'mobile'
    theTopHeader = shallowMount(TheTopHeader, {
      stubs: ['ny-level', 'ny-level-item', 'ny-icon'],
      mocks: {
        $mq
      }
    })
    expect(theTopHeader.vm.extraClasses).toBe('is-mobile')
  })

  test('should extraClasses to be empty', () => {
    const $mq = 'desktop'
    theTopHeader = shallowMount(TheTopHeader, {
      stubs: ['ny-level', 'ny-level-item', 'ny-icon'],
      mocks: {
        $mq
      }
    })
    expect(theTopHeader.vm.extraClasses).toBe('')
  })
})
