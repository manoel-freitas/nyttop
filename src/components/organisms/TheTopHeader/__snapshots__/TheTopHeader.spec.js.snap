// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheTopHeader Component snapshot test 1`] = `
<ny-level-stub
  extra-classes="is-mobile"
>
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
</ny-level-stub>
`;
